app/controllers/course/survey/responses_controller.rb

```diff
@@ -28,11 +28,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
```

Comment:
Must check the permission of `unsubmit`, otherwise student can actually send requests to unsubmit their own surveys.

This can be done through strong params, something like:
```
permitted_params = []
if can?(:update, @survey)
   permitted_params << :unsubmit
end
params.require(:response).permit(permmited_params)
```